python/ray/data/tests/test_actor_pool_fault_tolerance.py

```diff
@@ -0,0 +1,120 @@
+import asyncio
+import threading
+
+import pytest
+
+import ray
+from ray.tests.conftest import *  # noqa
+
+
+def test_removed_nodes_and_added_back(ray_start_cluster):
```

Comment:
* let's also test `pending_processor_usage` reports the correct usages during different stages. 
* maybe just move this test to `test_actor_pool_map_operator.py`.